accepted/0000-workspaces-part2.md

```diff
@@ -0,0 +1,211 @@
+
+```
+$ npm workspace core install react@16 --save-peer --workspace-group core=core/*
+```
+
+## Prior Art
+
+#### Previously
+
+- [npm workspaces](https://github.com/npm/rfcs/blob/de8d71c0453f5cf443d3ef2f47e313f12dd6aaf9/accepted/0000-workspaces.md)
+
+#### Filtering a subset of workspaces
+
+- [lerna filter-options](https://www.npmjs.com/package/@lerna/filter-options)
+- [Yarn v1 workspace](https://classic.yarnpkg.com/en/docs/cli/workspace)
+- [Yarn v2 foreach include/exclude](https://yarnpkg.com/cli/workspaces/foreach)
+- [pnpm recursive --filter](https://pnpm.js.org/en/cli/recursive#filter-lt-package_selector)
+
+## Dictionary
+
+During the discussions around this RFC it was brought up to our attention that a lot of the vocabulary surrounding what the larger JavaScript community understands as "workspaces" can be confusing, for the sake of keeping the discussion as productive as possible we're taking the extra step of documenting what each of the terms used here means:
+
+- **npm cli**: The [npm cli](https://github.com/npm/cli/) :wink:
+- **npm commands**: The set of **npm cli** [commands](https://docs.npmjs.com/cli-documentation/)
+- **npm workspaces**: The feature name, meaning the ability to the **npm cli** to support a better workflow for working with multiple packages.
+- **workspaces**: A set of **workspace**s.
+- **workspace**: A nested package within the **Top-level workspace** file system that is explicitly defined as such via **workspaces configuration**.
+- **Top-level workspace**: The root level package that contains a **workspaces configuration** defining **workspaces**.
+- **workspaces configuration**: The blob of json configuration defined within `package.json` that declares where to find **workspaces** for this **Top-level workspace** package.
+- **dependency**: A package that is depended upon by another given package.
+- **[peer dependency](https://docs.npmjs.com/files/package.json#peerdependencies)**: A special **dependency** relationship between packages.
+- **dependent**: A package which depends on another given package.
+- **symlink**: A [symbolic link](https://en.wikipedia.org/wiki/Symbolic_link) between files.
+- **[globs](https://en.wikipedia.org/wiki/Glob_(programming))**: String patterns that specifies sets of filenames with special characters.
+- **[Arborist](https://github.com/npm/arborist)**: The npm@7 install library
+- **hoisting packages**: Bringing packages up a level in the context of an installation tree.
+
+## Unresolved Questions and Bikeshedding
+
+- Filter syntax:
+    - `npm workspace foo test`
+    - `npm ws foo test`
+    - `npm :foo test`
+    - `npm --workspace=foo test`
+    - `npm -w=foo test`
+- Groups as a cli argument: `--workspace-group`
+    - how to define the `key=value` nature of it?
+    - is there any arg currently in the cli using key/value pairs?
+    - maybe it's not worth having it?
```

Comment:
@evocateur I'd really appreciate your input on this one, this idea was surfaced during the life of this RFC but I'm afraid it might be an invite to all the category of problems you listed with `--scope`

Reply:
I think it's better dropped as well. There's plenty of `package.json` config you can't change via the CLI. Aliases should be a little hard to change anyway, since there can be big consequences to accidentally changing them. In the context of a deploy, it can take down a site or break a published package.

Reply:
Aliases shouldn’t be able to be changed, but absolutely from the cli i need to be able to scope to subpackages on the fly. Very few things in package.json that *npm* looks at are not configurable from the cli.